### PR TITLE
INTLY-2816 - split up the browser tests execution

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
I don't say this is the best solution but I think it's more granular and more clear (once there is some failure)

**Comment by tremes** [comment]
Executed here https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/all/job/tremes-all-tests-executor/1/console for PoC testing.

File: jobs/integr8ly/ocp3/test-suites/all-tests-executor.yaml [hunk 1]
```diff
@@ -152,3 +167,13 @@
                 currentBuild.result = 'UNSTABLE'
             }
         }
+
+           def runSingleBrowserTest(testName) {
```
Comment (pawelpaszki) [inline jobs/integr8ly/ocp3/test-suites/all-tests-executor.yaml]: Please fix the inconsistent spacing

**Comment by pawelpaszki** [comment]
would it not be better to get the result of 00_walkthroughs test and don't execute wt-related tests in case of failure?

File: jobs/integr8ly/ocp3/test-suites/all-tests-executor.yaml (hunk 1 above)
Comment (tremes) [reply jobs/integr8ly/ocp3/test-suites/all-tests-executor.yaml]: oh yeah..thanks

**Comment by tremes** [comment]
@pawelpaszki interesting idea...maybe yes. Let's see what others think.

**Comment by damienomurchu** [comment]
I like the changes, and agree with suggestion to not run a wt-related test if the WT is not in the solution explorer. If one walkthrough is unavailable, but the others are, we should still test the run the WT test(s) for the WT(s) that are there though. 

That probably means splitting out `00_walkthroughs_start` as that currently tests for presence of all WT's.

**Comment by trepel** [comment]
To be honest I would leave this as is and don't do any skipping of other browser tests just because `00_walkthroughs` test failed.

If we want to make our life easier we can make sure the test fails early if some prereqs are not met. Ideally with a message like this one: `This test expects this and this to be in place but it seems it is not the case.` But this holds for each and every test, not just the browser ones.

**Comment by trepel** [comment]
We can skip the wt test if wt is not present - no issues with that. But in that case, we need to have separate test checking the presence of each walkthrough. So that there is some failed test in case some wt is missing (not just skipped test, that is easier to overlook).

**Comment by tremes** [comment]
I added one more commit. Did you mean something like that @pawelpaszki pls? And what others think please?

**Comment by trepel** [comment]
I think this one is good to go :)

**Comment by tremes** [comment]
In theory we could also skip the w1 and w2 executors if there's some failure in the start walkthroughs test. WDYT?

**Comment by pawelpaszki** [comment]
that was my original thought - to skip all wt1 and wt2 related (backend and ui) tests

**Comment by trepel** [comment]
Is there any (sub)test in wt1 or wt2 executor that would pass even if start walkthroughs test failed? I don't think so, but you might know. If yes, there would still be some value in running them.

Also, there is an option that start walkthroughs test fails at the very end but it does effectively start them - not highly probable but still.

I would slightly prefer to fail early saying the prereqs were not met instead of skipping.

**Comment by tremes** [comment]
The w2 and w1 executor tests require existence of related "shared services" namespace and also w1 or  w2 namespace. So there won't pass any test when these don't exist.

Second option can probably happen, but it's unlikely IMO (maybe some timeout could be likely). 

I don't have a strong opinion here. I can update it or leave as it is....as you want guys.

**Comment by trepel** [comment]
no strong opinion either. I guess it is you who is to do the actual work so up to you really :)

**Comment by tremes** [comment]
Let's be optimistic and believe that the start walkthrough test won't fail so often :) I am leaving it in current shape.

**Pull request merged**